### PR TITLE
Integration tests for Dependency Injection

### PR DESCRIPTION
**Pull request by himanshuc71**
Tests for: 
- Inject field non final diagnostic
- Inject method non final, non static, non abstract, non generic

Should close, #195, #221

### CODE REVIEW

**Comment by ananya2801** [comment]
Tests for the diagnostics look good. Do we need tests for the quick fix though?

**Review by ananya2801 — COMMENTED** [review]
Tests for the diagnostics look good. Do we need tests for the quick fixes though?

**Comment by kathrynkodama** [comment]
@AlvinTan2000 please review as well

**Comment by himanshuc71** [comment]
> Tests for the diagnostics look good. Do we need tests for the quick fixes though?

Added, please review, thanks.

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/di/DependencyInjectionTest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+         */
+        
+        // for d1
+        JakartaJavaCodeActionParams codeActionParams = createCodeActionParams(uri, d1);
+        TextEdit te = te(27, 4, 28, 4,
+                "");
+        CodeAction ca = ca(uri, "Remove @Inject", d1, te);
+        TextEdit te1 = te(28, 11, 28, 17,
+                "");
+        CodeAction ca1 = ca(uri, "Remove final modifier from this field", d1, te1);
```
Comment (kathrynkodama) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/di/DependencyInjectionTest.java:77]: Add quotations around 'field', see the corresponding test error: 
```
DependencyInjectionTest.DependencyInjectionDiagnostics:78 Assert title [1] expected:<Remove [final] modifier from this ...> but was:<Remove [the 'final'] modifier from this..>
```

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/projects/jakarta-sample/src/main/java/io/openliberty/sample/jakarta/di/GreetingServlet.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+
+@WebServlet(name = "greetingServlet", urlPatterns = { "/di" })
+public abstract class GreetingServlet extends HttpServlet {
+
+    /**
+     *
+     */
+private static final long serialVersionUID = 1L;
```
Comment (AlvinTan2000) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/projects/jakarta-sample/src/main/java/io/openliberty/sample/jakarta/di/GreetingServlet.java:25]: Indent looks off

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/di/DependencyInjectionTest.java (hunk 1 above)
Comment (himanshuc71) [reply jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/di/DependencyInjectionTest.java]: fixed

**Review by kathrynkodama — APPROVED** [review]
LGTM, thanks @himanshuc71

**Pull request merged**